client/src/store/ticket.js

```diff
@@ -0,0 +1,122 @@
+import { observable, action } from 'mobx';
+
+import axios from '../config';
+import { notification } from 'antd';
+
+class TicketStore {
+    @observable tickets = [];
+    @observable loading = false;
+
+    @action
+    getTickets = async (data) => {
```

Comment:
Three things
- All CURD mentioned here has duplicated code that we can extract and use in more generic way, like set loading, showing error response
- If we set async/await for API calls, we should use always async/await for the places where we do something after calling this method, would be better just return promise
- In most case (especially using DRF), the endpoints action URL will be same for CRUD, so there will be only difference in endpoints name like tickets, ticket_purchases  etc.., In case we can just add endpoint in constructor and use it for all other API call, (for building generic API view), and then we can extend it for all other data stores

ex: 
return new Promise((resolve, reject) => {
      axios.post('/tickets', data)
        .then((response) => {
          this.successfulPost(response ? response.data : {})
          resolve(response)
        })
        .catch((error) => {
          this.failedPost()
          reject(error)
        })
    })